src/python/tritonfrontend/CMakeLists.txt

```diff
@@ -0,0 +1,124 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cmake_minimum_required (VERSION 3.18)
+
+# ================= Ensures Package is Structured Properly ==================
+# Top level module entry point and typed marker
+file(COPY __init__.py DESTINATION .)
+file(COPY py.typed DESTINATION .)
+# Copy the '__init__.py' for the '_c' module
+file(COPY _c/__init__.py DESTINATION ./_c/.)
+file(COPY _c/__init__.pyi DESTINATION ./_c/.)
+
+# Find and copy _api modules
+file(GLOB PYTHON_MODULE_FILES ./_api/*.py)
+file(COPY ${PYTHON_MODULE_FILES} DESTINATION ./_api/.)
+# ================================= END =====================================
+
+
+# =================== Downloading and Installing pybind11 ===================
+include(FetchContent)
+
+FetchContent_Declare(
+    pybind11
+    GIT_REPOSITORY https://github.com/pybind/pybind11.git
+    GIT_TAG v2.13.1
+    GIT_SHALLOW ON
+)
+
+FetchContent_MakeAvailable(pybind11)
+
+# ================================= END =====================================
+
+
+find_package(CUDAToolkit REQUIRED)
+
+add_library(
+  python-binding-dependency
+  EXCLUDE_FROM_ALL
+  ../../shared_memory_manager.h
+  ../../shared_memory_manager.cc
+  ../../data_compressor.h
+  ../../common.h
+  ../../common.cc
+  ../../restricted_features.h
+  ../../tracer.h
+  ../../classification.cc
+)
+
+target_link_libraries(
+  python-binding-dependency
+  PUBLIC
+  http-endpoint-library
+  grpc-endpoint-library
+  CUDA::cudart
```

Comment:
Dependency has been broken into conditional links that only link when the respective flags are marked as `ON`: `TRITON_ENABLE_HTTP`, `TRITON_ENABLE_GRPC`, and `TRITON_ENABLE_GPU`

Reply:
And since `tritonfrontend_pybind.cc` needs access to `HTTPServer`, `HTTPAPIServer`, and `grpc::Server` to pass as template arguments to `TritonFrontend`, it needs to link these libraries to have access to their respective implementations.